src/core/border_router/routing_manager.cpp

```diff
@@ -1129,12 +1129,17 @@ bool RoutingManager::AddDiscoveredPrefix(const Ip6::Prefix &aPrefix,
 
         SuccessOrExit(AddExternalRoute(aPrefix, aRoutePreference));
 
-        if (aIsOnLinkPrefix)
+        if (aIsOnLinkPrefix && mRouterSolicitCount > 0)
         {
             // Stop Router Solicitation if we discovered a valid on-link prefix.
             // Otherwise, we wait till the Router Solicitation process times out.
             // So the maximum delay before the Border Router starts advertising
-            // its own on-link prefix is 9 (4 + 4 + 1) seconds.
+            // its own on-link prefix is 10 seconds = RS_DELAY (1) + RS_INTERNAL (4)
+            // + RS_INTERVAL (4) + RS_DELAY (1).
+            //
+            // Per section 6.3 of RFC 4861, a host SHOULD send at least one
+            // solicitation in the case where an advertisement is received
+            // prior to having sent a solicitation.
```

Comment:
The purpose of this is because: 
```
   Responses to solicited advertisements may contain more information
   than unsolicited advertisements.
```
Are there situations where we might be missing information if we do not send the Router Solicitation?

Reply:
Consider that there are multiple border routers on the infra link, we may miss RIOs of other BRs if an unsolicited RA is received and we stop sending RS immediately. Always sending RS so that every RSs will response. Thoughts?

Reply:
Yes, that makes sense. However, I think this is a different reason than what the comment suggests. How about updating the comment to reflect the true purpose of requiring an RS?

Reply:
@jwhui Thanks for the suggestion! I updated the comments accordingly, PTAL.